service/storage_proxy.cc

```diff
@@ -2929,7 +2929,11 @@ storage_proxy::storage_proxy(distributed<replica::database>& db, storage_proxy::
         sm::make_queue_length("current_throttled_writes", [this] { return _throttled_writes.size(); },
                        sm::description("number of currently throttled write requests")),
     });
-
+    _metrics.add_group(storage_proxy_stats::REPLICA_STATS_CATEGORY, {
+        sm::make_current_bytes("view_update_backlog", [this] { return _max_view_update_backlog.fetch_shard(this_shard_id()).get_current_bytes(); },
+                       sm::description("View update backlog size used for slowing down writes when it grows too large"
+                                       "Should closely resemble database_view_update_backlog")),
```

Comment:
I think this is the third (?) view-backlog metric we have. I think you should explain the difference, not that it "should closely resemble". Perhaps say that this one is a per-shard metric, and also I think you shouldn't say that this one is "used for slowing down writes" because what is used for slowing down writes is calculated on the coordinator, not the replica, and definitely not per shard.

Reply:
I can update the description, though it will need to be quite verbose to capture the difference between this and the `database_view_update_backlog` metric (in particular, that one is also per-shard). Maybe I could also make this into an API, assuming we have no more bugs in the backlog updating code, it's practically duplicating the `database_view_update_backlog` and we only need it for the test

Reply:
We already have the `view_update_backlog` metric, which is per-shard. Perhaps we could fetch it and compute the maximum of all shards' values ourselves?

Reply:
For now, in the rebase, I added a more descriptive comment. We can't use the `scylla_database_view_update_backlog` for the test, we're in fact testing that a different variable(`_max_view_update_backlog`) changes after the variable represented by `scylla_database_view_update_backlog` changes. Perhaps we could unify them into one, but at least in the current implementation, that would require us to make an atomic semaphore, which I'm not sure is the most efficient

Reply:
The new comment is better, and I am finding it hard to reject the new metric, so I won't reject it. But in the big picture, it's still not a great thing that we have different metrics with similar meanings that ordinary users will probably not be able to make sense of, and I'm not sure it's great that we add metrics just because they are convenient for tests and not because they are convenient for users.
Maybe we should consider adding "debug-only" metrics, which are visible in tests but not for ordinary users. But we can consider this later.